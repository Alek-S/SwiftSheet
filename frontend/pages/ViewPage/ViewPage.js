import React from 'react';
import styled from 'styled-components';
import defaultStyle from '../../defaultStyle';
import RecentlyViewed from '../../components/RecentlyViewed/RecentlyViewed';
import SearchSheet from '../../components/SearchSheet/SearchSheet';
import ReactGA from 'react-ga';

const ViewPage = () => {
	ReactGA.pageview('/view');

	return (
		<StyledDiv>
			<SearchSheet />
			<RecentlyViewed />
		</StyledDiv>
	);
};

const StyledDiv = styled(defaultStyle)`
	padding-top: 6rem;
	background-color: ${props => props.theme.color.background};
	height: calc(100vh - 80px);
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
`;

export default ViewPage;
