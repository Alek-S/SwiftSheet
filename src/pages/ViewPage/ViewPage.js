import React from 'react';
import styled from 'styled-components';
import defaultStyle from '../../defaultStyle';

const StyledDiv = styled(defaultStyle)`
	padding-top: 20rem;
`;

const ViewPage = () => <StyledDiv>View Page</StyledDiv>;

export default ViewPage;
