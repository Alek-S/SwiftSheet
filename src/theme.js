/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
