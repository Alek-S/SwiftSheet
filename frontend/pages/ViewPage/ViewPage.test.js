import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';

import ViewPage, { StyledDiv } from './ViewPage';
import theme from '../../theme';

describe('ViewPage', () => {
	test('matches the snapshot', () => {
		const tree = shallow(<ViewPage theme={theme} />);
		expect(toJson(tree)).toMatchSnapshot();
	});

	test('StyledDiv matches the snapshot', () => {
		const tree = renderer.create(<StyledDiv theme={theme} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
