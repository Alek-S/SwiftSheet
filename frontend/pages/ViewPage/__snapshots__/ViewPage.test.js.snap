// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewPage StyledDiv matches the snapshot 1`] = `
.c0 {
  background-color: white;
  color: #404040;
  font-family: 'Quicksand',sans-serif;
  font-size: 0.9em;
  padding-top: 6rem;
  background-color: #fafafa;
  height: calc(100vh - 80px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;

exports[`ViewPage matches the snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "ViewPage__StyledDiv-wujrxm-0",
        "isStatic": false,
        "rules": Array [
          "background-color:white;color:",
          [Function],
          ";font-family:",
          [Function],
          ";font-size:0.9em;",
          "padding-top:6rem;background-color:",
          [Function],
          ";height:calc(100vh - 80px);display:flex;flex-direction:column;justify-content:center;align-items:center;",
        ],
      },
      "displayName": "ViewPage__StyledDiv",
      "foldedComponentIds": Array [
        "defaultStyle-g21vfe-0",
      ],
      "render": [Function],
      "styledComponentId": "ViewPage__StyledDiv-wujrxm-0",
      "target": "div",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <SearchSheet />
  <RecentlyViewed />
</StyledComponent>
`;
